lib/chef/log.rb

```diff
@@ -34,6 +35,14 @@ def self.show_time=(*args)
       end
     end
 
+    def self.deprecation(msg=nil, &block)
+      if Chef::Config[:treat_deprecation_warnings_as_errors]
+        error(msg, &block)
+        raise Chef::Exceptions::DeprecatedFeatureError
```

Comment:
passed the message.

array arguments to chef/log look like this right now:

```
[2015-02-09T17:00:27-08:00] WARN: ["foo", "bar"]
```

if msg is an array right now this code looks like:

```
[2015-02-09T17:00:50-08:00] ERROR: ["foo", "bar"]
[2015-02-09T17:00:50-08:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
[2015-02-09T17:00:50-08:00] FATAL: Chef::Exceptions::DeprecatedFeatureError: ["foo", "bar"]
```

which is consistent with the other methods.